src/index.ts

```diff
@@ -7,24 +7,103 @@ const points = new Array<{ x: number, y: number, segmentLength: number }>();
 let unclosedLength = 0;
 let unclosedPath = new Path2D();
 
-const fftSize = 4096;
-const fft = new FFT(fftSize);
-const input = fft.createComplexArray() as number[];
-const output = fft.createComplexArray() as number[];
+let fftSize = 4096;
+let fft = new FFT(fftSize);
+let input = fft.createComplexArray() as number[];
+let output = fft.createComplexArray() as number[];
```

Comment:
When a set of related variables become mutable, I would eventually like to encapsulate them. I don't have a fully-formed thought on this yet though. One goal will be to reduce the duplication that is being introduced between what these lines are doing and what `loadLocation` does.

Reply:
(Leaving a comment after conversation resolution)
I thought those could go inside the `initControls` function; that raised errors, I then decided they were perfectly fine as-is.
But one day, they might go somewhere better thought. :)